File: src/keyboard.py
import fileinput
import sys

if len(sys.argv) > 1:
    layout = sys.argv[1]
else:
    layout = "us"

with fileinput.input("/etc/default/keyboard", inplace=True) as file:
    for line in file:
        if "XKBLAYOUT" in line:
            print('XKBLAYOUT="' + layout + '"')
        else:
            print(line, end="")

File: src/keyboard.rs
use std::env;
use std::fs;

const PATH: &str = "/etc/default/keyboard";

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() < 2 {
        panic!("Not enough arguments");
    }

    let layout = &args[1];
    let contents = fs::read_to_string(PATH).unwrap();

    let result: String = contents.lines()
        .map(|line| {
            let mut line = String::from(line);

            if line.contains("XKBLAYOUT") {
                line = String::from("XKBLAYOUT=\"");
                line.push_str(layout);
                line.push('"');
            }

            line.push('\n');
            line
        }).collect();

    fs::write(PATH, result).unwrap();
}
